apps/edit/app.lisp

```diff
@@ -69,49 +67,49 @@
 
 ;cursor_xy = (+ (mouse_xy / char_wh) offset_xy)
 (defun-bind mouse-cursor (mouse_xy)
-	(defq buffer (elem text_buffer current_text))
-	(bind '(ox oy cx cy sx) (elem text_position current_text))
+	(defq buffer (elem (const text_buffer) current_text))
+	(bind '(ox oy cx cy sx) (elem (const text_position) current_text))
 	(defq cursor_xy (list cx cy) char_wh (vdu-char-size vdu) offset_xy (list ox oy))
 	(setq cursor_xy (map + (map / mouse_xy char_wh) offset_xy)
 		cx (elem 0 cursor_xy) cy (elem 1 cursor_xy))
 	(if (>= cy (length buffer)) 
 		(setq cy (min cy (dec (length buffer))) cx (length (elem cy buffer))))
 	(if (> cx (length (elem cy buffer)))
 		(setq cx (max (set-sticky) (length (elem cy buffer)))) (setq sx cx))
-	(elem-set text_buffer current_text buffer)
-	(elem-set text_position current_text (list ox oy cx cy sx)))
+	(elem-set (const text_buffer) current_text buffer)
+	(elem-set (const text_position) current_text (list ox oy cx cy sx)))
 
```

Comment:
(elem-set (const text_buffer) current_text buffer)

I've been thinking about this (const) use for tupple indexes and think there is now a good case for a macro ! That macro would be somthing like (tupple-set text_buffer current_text buffer) and it would do the addition of the const. What do you think it seams it's happening all over, I do it and maybe this macro should go in the boot.inc file for everyone to use.

(defmacro-bind tupple-set (i l v)
      `(elem-set (const ,i) ,l ,v))

??

Reply:
Maybe also the (tupple-get) as well ? Again adds the (const) wrapper.

(defmacro-bind tupple-get (i l)
`(elem (const ,i) ,l))